src/robot/running/keywords.py

```diff
@@ -307,6 +328,35 @@ def _to_number_with_arithmetics(self, item):
         return number
 
 
+class ForInZipLoop(BasicForLoop):
+
+    def __init__(self, forstep, templated=False):
+        BasicForLoop.__init__(self, forstep, templated)
+
+    def _flavor_name(self):
+        return 'IN ZIP'
+
+    def _transform_items(self, items):
+        answer = list()
+        for zipped_item in zip(*items):
+            answer.extend(zipped_item)
+        return answer
+
+
+
+FOR_LOOP_CLASS_FROM_TYPE = dict(
+    INRANGE=ForInRangeLoop,
+    INZIP=ForInZipLoop,
+    )
+
+def ForLoop(forstep, templated=False):
+    try:
+        answer_class = FOR_LOOP_CLASS_FROM_TYPE[forstep.flavor]
+    except KeyError:
+        answer_class = BasicForLoop
```

Comment:
I took your `get(flavor, BasicForLoopRunner)` suggestion but didn't inline the dict yet.

Would it be totally crazy to allow libraries to add new kinds of for-loops by defining their own `BasicForLoopRunner` subclass and calling a method that adds it to the factory's dict?